### PR TITLE
Cluster Templates

### PR DESCRIPTION
**Pull request by sean-smith**
## Description

* Pre-Defined Cluster templates for common applications
* All templates are hosted in `/templates` directory on Github

## Demo

https://user-images.githubusercontent.com/5545980/178086496-5946f18d-d99d-4f3f-96a9-263c76815dbc.mov

## Testing

Change the path of the Github URL in [api/PclusterApiHandler.py#210](https://github.com/aws-samples/pcluster-manager/pull/179/files#diff-5424a9a7f7cf5ddc8d3e420f077ac509a3b37f81c8cbf7c9bac5f8695fc9a8e2)

**from**
```bash
https://raw.githubusercontent.com/aws-samples/pcluster-manager/main/{template_path}
```

**to**
```bash
https://raw.githubusercontent.com/sean-smith/pcluster-manager/template/{template_path}
```

Signed-off-by: Sean Smith <seaam@amazon.com>

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -7,3 +7,5 @@ infrastructure/cognitolambda/node_modules
 .DS_Store
 .hugo_build.lock
 public/*
+package.json
+package-lock.json
```
Comment (mendaomn) [inline .gitignore:11]: issue(blocking): Careful, these two cannot be gitignored :)

File: app.py [hunk 2]
```diff
@@ -94,6 +95,11 @@ def ec2_action_():
     def get_cluster_config_():
         return get_cluster_config()
 
+    @app.route("/manager/get_cluster_template")
```
Comment (mtfranchetto) [inline app.py:98]: suggestion: cluster templates are plain strings and can probably be inlined in the JS bundle as they add few extra bytes (nothing comparable to third party modules). What I suggest is to remove this extra configuration management and use something like [next-plugin-yaml](https://www.npmjs.com/package/next-plugin-yaml) to inline the templates frontend side. This has several benefits:
- no new route backend side
- no new custom requests frontend side and error management
- no manual transform of yaml -> js: this gets handled automatically by `next-plugin-yaml`
- simplified testing: no need to manually edit URL to fetch template, simply edit the files and see the result in the browser

File: frontend/src/old-pages/Configure/Source.js [hunk 3]
```diff
@@ -181,7 +194,13 @@ function Source() {
               {
                 value: "wizard",
                   label: "Wizard",
-                  description: "Choose this to start a new cluster configuration."
+                  description: <Box margin={{bottom: "xs"}} >
```
Comment (mtfranchetto) [inline frontend/src/old-pages/Configure/Source.js:197]: Screenshot 2022-07-11 at 11 39 18
issue: I personally find very confusing this kind of UX where I can pick a template from the list but I also have another option called in the same way (Template) beneath it. While this is not a major issue it still needs some UX attention, and can be reworked in different ways:
- rename the Template section: maybe with a less fancy name like "Wizard with existing configuration"?
- rename the new templates feature: presets?
- add another option for our users: Empty wizard / Wizard with preset / Wizard from existing configuration / and so on

@sean-smith what do you think?

**Review by mtfranchetto — CHANGES_REQUESTED** [review]

File: app.py (hunk 2 above)
Comment (sean-smith) [reply app.py:98]: I like this approach! I think the only downside is we can't update between releases then right?

File: frontend/src/old-pages/Configure/Source.js (hunk 3 above)
Comment (sean-smith) [reply frontend/src/old-pages/Configure/Source.js:197]: I think having two Wizard sections might be confusing but I could re-work the template section to be either "Start from an existing template" or "upload your own template" maybe with a toggle UI element, or a setting in the list "Upload a default template". Definitely want some PM eyes on this feature.

File: app.py (hunk 2 above)
Comment (mtfranchetto) [reply app.py:98]: Yes, correct, but we can start small by making those files static and then, if we have to for some reasons, write some code to add dynamic imports from cloud. Anyway, I think that if we invest in testing, build times and automated deployments it is going to be much easier to release something thus static files should be ok imho.

**Pull request closed without merging**